tools/tdbg/commands.go

```diff
@@ -441,20 +471,20 @@ func AdminListShardTasks(c *cli.Context, clientFactory ClientFactory) error {
 }
 
 // AdminRemoveTask describes history host
-func AdminRemoveTask(c *cli.Context, clientFactory ClientFactory) error {
+func AdminRemoveTask(
+	c *cli.Context,
+	clientFactory ClientFactory,
+	taskCategoryRegistry tasks.TaskCategoryRegistry,
+) error {
 	adminClient := clientFactory.AdminClient(c)
 	shardID := c.Int(FlagShardID)
 	taskID := c.Int64(FlagTaskID)
-	categoryInt, err := StringToEnum(c.String(FlagTaskType), enumsspb.TaskCategory_value)
+	category, err := getCategory(taskCategoryRegistry, c.String(FlagTaskType))
 	if err != nil {
-		return fmt.Errorf("unable to parse Task Type: %s", err)
-	}
-	category := enumsspb.TaskCategory(categoryInt)
-	if category == enumsspb.TASK_CATEGORY_UNSPECIFIED {
-		return fmt.Errorf("task type %s is currently not supported", category)
+		return err
 	}
 	var visibilityTimestamp int64
-	if category == enumsspb.TASK_CATEGORY_TIMER {
+	if category == tasks.CategoryTimer {
```

Comment:
need to check if category type is scheduled. We will need the visibility timestamp for all scheduled queues.

Reply:
Done